src/tunnelbridge_cmd.cpp

```diff
@@ -528,12 +542,24 @@ CommandCost CmdBuildBridge(TileIndex end_tile, DoCommandFlag flags, uint32 p1, u
 		DiagDirection dir = AxisToDiagDir(direction);
 
 		Company *c = Company::GetIfValid(company);
+
+		if (!Bridge::CanAllocateItem()) {
+			return_cmd_error(STR_ERROR_TOO_MANY_BRIDGES);
+		}
+
+		Bridge *b = new Bridge();
+		b->type = bridge_type;
+		b->build_date = _date;
+		b->town = CalcClosestTownFromTile(tile_start);
```

Comment:
This implies that otherwise identical bridges could have a different assigned town depending on which end is the "start" tile.

If a town is removed (e.g. in the scenario editor) the town pointer could be left dangling. Likewise if a town is added this could no longer be the closest town.
If neither end is actually in a town, should this be set at all?

Reply:
It is all same to stations, objects, etc.(which have towns as 'related objects')... I just copy-pasted it from objects.

Reply:
But which tile the town is taken from is mostly well defined. For a bridge it should maybe always be e.g. the northern end tile, no matter which is "start" or "end".

Stations, objects etc are explicitly handled in `CmdDeleteTown`.

Reply:
In tunnelbridge_cmd.cpp, line 336, tile_start and tile_end are swapped in order to make tile_start to always be the northen end tile, so it isn't problem.

I updated code, now bridges are handled in `CmdDeleteTown`.